PhysicsTools/HeppyCore/python/framework/event.py

```diff
@@ -1,47 +1,124 @@
+import pprint
+import copy
 import collections
+import fnmatch
+
 from ROOT import TChain
 
 class Event(object):
     '''Event class.
 
-    The Looper passes the Event object to each of its Analyzers,
+    The Looper passes an Event object to each of its Analyzers,
     which in turn can:
     - read some information
     - add more information
     - modify existing information.
 
-    Attributes:
-      iEv = event processing index, starting at 0
-      eventWeight = a weight, set to 1 at the beginning of the processing
-      input = input, as determined by the looper
+    A printout can be obtained by doing e.g.:
+    
+      event = Event() 
+      print event 
+
+    The printout can be controlled by the following class attributes:
+      print_nstrip : number of items in sequence to be printed before stripping the following items
+      print_patterns : list of patterns. By default, this list is set to ['*'] so that all attributes are
+                    printed
+
+    Example:
+      event = Event()
+      Event.print_nstrip = 5  # print only the 5 first items of sequences
+      Event.print_patterns = ['*particles*', 'jet*'] # only print the attributes that 
+                                                     # contain "particles" in their name or
+                                                     # have a name starting by "jet" 
+       
+    Object attributes:
+      iEv: event processing index, starting at 0
+      eventWeight: a weight, set to 1 at the beginning of the processing
+      input: input, as determined by the looper
+      analyzers: list of analyzers that processed this event, with their result, in the form:
+          [(analyzer_name, result?), ...]
     #TODO: provide a clear interface for access control (put, get, del products) - we should keep track of the name and id of the analyzer.
     '''
 
-    def __init__(self, iEv, input_data=None, setup=None, eventWeight=1 ):
+    print_nstrip = 10
+    print_patterns = ['*']
+
+    def __init__(self, iEv, input_data=None, setup=None, eventWeight=1):
         self.iEv = iEv
         self.input = input_data
         self.setup = setup
         self.eventWeight = eventWeight
+        self.analyzers = []
+
+    def _get_print_attrs(self, subname=""):
+        '''returns a dict of printable information of an event
+        arguments
+        * subname is used when called recursively and is the name of the parent object'''
+        selected_attrs = copy.copy(self.__dict__) #initial selection of what we can print
+        selected_attrs.pop('setup') #get rid of some bits
+        selected_attrs.pop('input')
+        
+        # Colin: the following are unused: 
+        matched_attrs = dict() #this applies pattern matching to obtain a subset of selected_attrs
+        print_attrs=dict() #ready for printing
```

Comment:
if they're unused then why add them?

Reply:
hehe indeed. I guess this is a remnant of the prototyping of the code. removed.